lib/retriable.rb

```diff
@@ -73,4 +69,24 @@ def retriable(opts = {})
       end
     end
   end
+
+  def matched_exception?(on, exception)
+    return true unless on.is_a?(Hash)
+
+    on.any? do |expected_exception, matchers|
+      next false unless exception.is_a?(expected_exception)
+      next true if matchers.nil?
+
+      Array(matchers).any? do |matcher|
+        if matcher.is_a?(Regexp)
+          exception.message =~ matcher
+        elsif matcher.is_a?(Proc)
+          matcher.call(exception)
```

Comment:
would it be worth passing any additional params to the `Proc`?  Maybe like [the `on_retry.call()`](https://github.com/kamui/retriable/blob/7b9da542e89d9e2ffaf4f1803fc34128a2f806e7/lib/retriable.rb#L70)